# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/windowwatcher/nsWindowWatcher.h
# Commit: fde2d124a1b1
# Full Hash: fde2d124a1b15877570390c7c69f5fcf6770640d
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-08 03:36:50
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 7.2: Create and use nsOpenWindowInfo types in nsWindowWatcher logic, r=kmag
#   
#   This patch builds on top of part 7.1 by creating this object within
#   nsWindowWatcher and ContentParent to carry the relevant information through
#   provider interfaces when creating new content windows. The nsOpenWindowInfo
# ==============================================================================

diff -r 7ccbbb127b7f -r fde2d124a1b1 toolkit/components/windowwatcher/nsWindowWatcher.h
--- a/toolkit/components/windowwatcher/nsWindowWatcher.h	Tue Apr 07 21:39:24 2020 +0000
+++ b/toolkit/components/windowwatcher/nsWindowWatcher.h	Tue Apr 07 21:39:32 2020 +0000
@@ -20,6 +20,7 @@
 #include "mozilla/Maybe.h"
 #include "nsIWindowCreator.h"  // for stupid compilers
 #include "nsIWindowWatcher.h"
+#include "nsIOpenWindowInfo.h"
 #include "nsIPromptFactory.h"
 #include "nsIRemoteTab.h"
 #include "nsPIWindowWatcher.h"
@@ -115,9 +116,7 @@
   nsresult CreateChromeWindow(const nsACString& aFeatures,
                               nsIWebBrowserChrome* aParentChrome,
                               uint32_t aChromeFlags,
-                              nsIRemoteTab* aOpeningBrowserParent,
-                              mozIDOMWindowProxy* aOpener,
-                              uint64_t aNextRemoteTabId,
+                              nsIOpenWindowInfo* aOpenWindowInfo,
                               nsIWebBrowserChrome** aResult);
 
   void MaybeDisablePersistence(const nsACString& aFeatures,